tchannel/serializer/thrift.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from thrift.protocol import TBinaryProtocol
+from thrift.transport import TTransport
+
+from tchannel.schemes import THRIFT
+
+from .. import io
+from .. import rw
+
+
+class ThriftSerializer(object):
+
+    _headers_rw = rw.headers(
+        rw.number(2),
+        rw.len_prefixed_string(rw.number(2)),
+        rw.len_prefixed_string(rw.number(2)),
+    )
+
+    def __init__(self, deserialize_type):
+        self.name = THRIFT
+        self.deserialize_type = deserialize_type
```

Comment:
This is only used in deserialize_body, make it a parameter to that method, and then you can make all these serializers pure functions. In fact, the serialize_header, then serialize_body thing is a bit unessecary, since you'll always want to perform these operations at the same time. Suggest:

```py
from serializers.json import serialize, deserialize

headers, body = serialize(headers, body)

#...

headers, body = deserialize(headers, body)
```